### ISSUE TITLE
S3 will not return '/' in CommonPrefixes when there is.

### ISSUE THREAD

**Issue by vuryleo** [spulec/moto#2759]
## Reporting Bugs

# Description
When there is some Key with consecutive slashes `'/'`, `list_objects_v2` with `Delimiter='/'` will not include the right answer in `CommonPrefixes`

# Code to reproduce
```python
import boto3
from moto import mock_s3

@mock_s3
def list_slashes():
    c = boto3.client('s3')
    c.create_bucket(Bucket='bucket')
    c.put_object(Bucket='bucket', Key='///')

    resp = c.list_objects_v2(Bucket='bucket', Prefix='', Delimiter='/')
    print(resp)

    resp = c.list_objects_v2(Bucket='bucket', Prefix='/', Delimiter='/')
    print(resp)

list_slashes()
```

# Expected
```python
{
  ...,
 'CommonPrefixes': [{'Prefix': '/'}],
 'KeyCount': 1,
  ...,
}
{
  ...,
 'CommonPrefixes': [{'Prefix': '//'}],
 'KeyCount': 1,
  ...,
}
```

# Reality
```python
{
   ...,
   'Contents': [{'Key': '', ..., }],
   'KeyCount': 1,
   ...
}
{
   ...,
   'KeyCount': 0,
   ...,
}
```

sample output from real aws S3
```python
import boto3
c = boto3.client('s3')
c.put_object(Bucket='oss-playground', Key='///') 
print(c.list_objects_v2(Bucket='oss-playground', Prefix='//', Delimiter='/'))

```
Output
```python
{
 ...,
 'IsTruncated': False,
 'Name': 'oss-playground',
 'Prefix': '//',
 'Delimiter': '/',
 'CommonPrefixes': [{'Prefix': '///'}],
 'KeyCount': 1}
```

**Comment by oliparcol** [comment]
This issue seems to impact other functions (e.g. `download_file`): the leadings `/` are ignored:
```python
from moto import mock_s3
import boto3

@mock_s3 
def test(): 
    client = boto3.client('s3') 
    client.create_bucket(Bucket='bucket') 
    client.put_object(Bucket='bucket', Key='key') 
    print(client.head_object(Bucket='bucket', Key='/key')) 

test()
```

**Comment by oliparcol** [comment]
This may be related to #1637 (and associated fix #2398).
